docs_src/design/adr/0005-Service-Self-Config.md

```diff
@@ -0,0 +1,118 @@
+
+### Overrides
+Environmental variables override configuration or command line option values in the following ways
+- Environmental variables override local configuration; that is when configuration for a service is obtained from the local config file
+- Environmental variables override configuration values as they are pushed (self seeded) into the configuration service (Consul).  This override only occurs once (as the values are pushed / seeded into Consul from the service).  Once pushed, configuration values are used from Consul and any environmental value is ignored unless the -o/--overwrite flag is on.  The name of the environmental variable must match the path names in Consul.
+- Environmental variables can override command line options (except the -o overwrite and -r registry command line options). 
+
+NOTES:
+- Environmental variables do not override any local configuration; that is when configuration for a service is obtained from the local config file but not from the configuration service, the environmental variables are ignored.
+- Environmental variable overrides remove the need to change the "docker" profile in the res/docker/configuration.toml files - Allowing removal of 50% of the existing configuration.toml files.
+
+All values overriden get logged (indicating which configuration value or op param and the new value).  
+
+## Decision 
+
+These features have been implemented (with some minor changes to be done) for consideration here:  https://github.com/edgexfoundry/go-mod-bootstrap/compare/master...lenny-intel:SelfSeed2
+
+The implementation for self-seeding services and environmental overrides is already implemented (for Fuji) per this document in the application services and device services (and instituted in the SDKs of each).
+
+## Backward compatibility
+Several aspects of this ADR contain backward compatibility issues for the device service and application service SDKs.  Therefore, for the upcoming minor release, the following guidelines and expections are added to provide for backward compatibility.
+
+- --registry=<url> for Device SDK
+        
+If used with URL , assume previous version and use URL for --configProvider option and default to http as DS SDK does today. Also use URL for registry connection.
+
+- --registry (no ‘=’) and w/o --configProvider for both SDKs
+
+Assume for this release that one would never use registry w/o config provider.
+Use registry configuration from file for config provider as is done in Fuji. 
```

Comment:
The previous sentence says assume nobody would ever use --registry w/out a config provider, yet this next sentence says "Use registry configuration from file for config provider as is done in Fuji".

I think this could be re-worded to:

If a configProvider URI isn't specified, but --registry (w/out a URI) is specified, then the service will use the default config provider URI read from its local configuration file.

Reply:
@tonyespy , there isn't a `default config provider URI` in the local configuration, thus it has to use the Registry provider information that is in the local configuration file.

How about:
```
If a configProvider URI isn't specified, but --registry (w/out a URI) is specified, then the service will use the Registry provider information from its local configuration file for both Configuration and Registry providers.
```